var/spack/repos/builtin/packages/py-onnx/package.py

```diff
@@ -99,3 +99,8 @@ def setup_build_environment(self, env):
         env.set(
             "CMAKE_BUILD_DIR", join_path(self.stage.path, f"spack-build-{self.spec.dag_hash(7)}")
         )
+        # The absl code will use std::partial_ordering because of the compiler
+        # but it will fail to compile without setting standard to c++20
+        with when("%clang@19:"):
+            env.set("CXXFLAGS", "-std=c++20")
+            env.set("SPACK_CXXFLAGS", "-std=c++20")
```

Comment:
Was this PR tested? This context manager only works for directives at class scope. It doesn't work in functions, where we can access `self.spec` and have conditionals like:
```python
if self.spec.satisfies("%clang@19:"):
```